src-electron/generator/helper-zcl.js

```diff
@@ -730,24 +731,31 @@ function zcl_global_commands(options) {
  * @param {*} options
  * @returns Promise of attribute iteration.
  */
-function zcl_attributes(options) {
+async function zcl_attributes(options) {
   // If used at the toplevel, 'this' is the toplevel context object.
   // when used at the cluster level, 'this' is a cluster
-  let promise = templateUtil
-    .ensureZclPackageIds(this)
-    .then((packageIds) => {
-      if ('id' in this) {
-        // We're functioning inside a nested context with an id, so we will only query for this cluster.
-        return queryZcl.selectAttributesByClusterIdIncludingGlobal(
-          this.global.db,
-          this.id,
-          packageIds
-        )
-      } else {
-        return queryZcl.selectAllAttributes(this.global.db, packageIds)
-      }
-    })
-    .then((atts) => templateUtil.collectBlocks(atts, options, this))
+  let packageIds = await templateUtil.ensureZclPackageIds(this)
+  let attributes = ''
+  if ('id' in this) {
+    // We're functioning inside a nested context with an id, so we will only query for this cluster.
+    attributes = await queryZcl.selectAttributesByClusterIdIncludingGlobal(
+      this.global.db,
+      this.id,
+      packageIds
+    )
+    attributes = await upgrade.computeStorageTemplate(
+      this.global.db,
+      this.id,
+      attributes
+    )
+  } else {
+    attributes = await queryZcl.selectAllAttributes(this.global.db, packageIds)
+  }
+  if ('removeKeys' in options.hash) {
```

Comment:
The removeKeys bit was not there before, right?  Why is it needed now?

Reply:
was not sure. will remove, thanks